src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -362,20 +362,31 @@ private static final class ParallelResumer {
                         nodes.put(n, se);
                     } else {
                         LOGGER.warning(() -> "Could not find FlowNode for " + se + " so it will not be resumed");
+                        // TODO: Should we call se.getContext().onFailure()?
```

Comment:
Well, it's kind of interesting. Specifically in the case of a missing or corrupt `FlowNode`, there are two cases that I see:
1. If `FlowExecution.heads` contains the bad node, then everything gets handled in `CpsFlowExecution.onLoad`, we never read `program.dat`, we create placeholder nodes, the Pipeline fails, great.
2. If `FlowExecution.heads` does not contain the bad node, then the Pipeline loads, we attempt to resume it, various warnings get logged, and it hangs forever. The problem is that [this call in CpsStepContext](https://github.com/jenkinsci/workflow-cps-plugin/blob/a6f67a8c907b4f3699bd54b46271e0ecd3caccc4/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java#L368) throws right away, so the `CpsThread` never resumes with the outcome, and the Pipeline hangs. It seems like [this log message should at least be bumped to `WARNING`](https://github.com/jenkinsci/workflow-cps-plugin/blob/a6f67a8c907b4f3699bd54b46271e0ecd3caccc4/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java#L438) because any errors there are likely to be the proximate cause of a build hanging forever, but I wonder if we should also attempt to call `CpsFlowExecution.croak` or `CpsVmExecutorService.reportProblem` in that catch block.

So long story short, it seems like there is no point trying to do anything special in these cases in this plugin.

Reply:
https://github.com/jenkinsci/workflow-cps-plugin/pull/916 updates the relevant log message to `WARNING` so that we have an idea if the problematic situation is ever occurring in practice.